rotkehlchen/globaldb/upgrades/v12_v13.py

```diff
@@ -0,0 +1,94 @@
+            INSERT OR IGNORE INTO token_kinds(token_kind, seq) VALUES ('E', 5);
+        """)
+
+    @progress_step('Add solana tokens table and populate from CSV')
+    def _add_and_populate_solana_tokens(write_cursor: 'DBCursor') -> None:
+        """Create solana tokens table and update asset identifiers to use new solana format"""
+        write_cursor.execute("""
+        CREATE TABLE IF NOT EXISTS solana_tokens (
+            identifier TEXT PRIMARY KEY NOT NULL COLLATE NOCASE,
+            token_kind CHAR(1) NOT NULL DEFAULT('D') REFERENCES token_kinds(token_kind),
+            address VARCHAR[44] NOT NULL,
+            decimals INTEGER,
+            protocol TEXT,
+            FOREIGN KEY(identifier) REFERENCES assets(identifier) ON UPDATE CASCADE ON DELETE CASCADE
+        )""")  # noqa: E501
+
+        dir_path = Path(__file__).resolve().parent.parent.parent
+        if (csv_file := dir_path / 'data' / 'solana_tokens_data.csv').exists() is False:
+            log.error(f'Missing required CSV file at {csv_file}. This should not happen. Bailing...')  # noqa: E501
+            return
+
+        asset_updates, solana_tokens_data = [], []
+        duplicated_tokens = ('TRISIG', 'HODLSOL')  # TRISIG maps to TRISG & HODLSOL maps to HODL
+        with csv_file.open(encoding='utf-8') as f:
+            reader = csv.DictReader(f)
+            for row in reader:
+                if (old_id := row['old_id']) in duplicated_tokens:  # skip duplicate mapping
+                    continue
+
+                asset_updates.append(((new_id := f'solana/token:{row["address"]}'), old_id))
+                solana_tokens_data.append((
+                    new_id,
+                    'D',  # spl token
+                    row['address'],
+                    row['decimals'],
+                    None,
+                ))
+
+        write_cursor.switch_foreign_keys('OFF')
+        write_cursor.executemany('INSERT INTO solana_tokens(identifier, token_kind, address, decimals, protocol) VALUES(?, ?, ?, ?, ?)', solana_tokens_data)  # noqa: E501
+        write_cursor.executemany('UPDATE assets SET identifier = ? WHERE identifier = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE common_asset_details SET identifier = ? WHERE identifier = ?', asset_updates)  # noqa: E501
+
+        # duplicates were skipped during `solana_tokens` creation to avoid unique constraint errors
+        # but we still need to update any existing foreign key references to these old identifiers
+        asset_updates.extend([
+            ('solana/token:BLDiYcvm3CLcgZ7XUBPgz6idSAkNmWY6MBbm8Xpjpump', 'TRISIG'),
+            ('solana/token:58UC31xFjDJhv1NnBF73mtxcsxN92SWjhYRzbfmvDREJ', 'HODLSOL'),
+        ])
+        write_cursor.executemany('UPDATE asset_collections SET main_asset = ? WHERE main_asset = ?', asset_updates)  # noqa: E501
+        write_cursor.executemany('UPDATE multiasset_mappings SET asset = ? WHERE asset = ?', asset_updates)  # noqa: E501
```

Comment:
mmm one thing. In the `assets` table of the user you also need to remap those assets right?